color/src/temperature.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 the Color Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![expect(
+    clippy::excessive_precision,
+    reason = "We are sticking with the constants from Tanner Helland."
+)]
+
+use crate::{AlphaColor, ColorSpace, OpaqueColor};
+
+#[cfg(all(not(feature = "std"), not(test)))]
+use crate::floatfuncs::FloatFuncs;
+
+// Implementation notes:
+//
+// * These can't be const yet as `clamp`, `ln`, and `powf` are not `const`.
+
+impl<CS: ColorSpace> AlphaColor<CS> {
+    /// Convert a temperature in Kelvin to a color.
+    ///
+    /// The Kelvin temperature will be clamped to a range of 1000-40,000.
+    pub fn from_kelvin(kelvin: f32) -> Self {
+        let [r, g, b] = CS::from_linear_srgb(from_kelvin(kelvin));
+        Self::new([r, g, b, 1.])
+    }
+}
+
+impl<CS: ColorSpace> OpaqueColor<CS> {
+    /// Convert a temperature in Kelvin to a color.
+    ///
+    /// The Kelvin temperature will be clamped to a range of 1000-40,000.
+    pub fn from_kelvin(kelvin: f32) -> Self {
+        Self::new(CS::from_linear_srgb(from_kelvin(kelvin)))
+    }
+}
```

Comment:
Perhaps the constructor should just be defined on the base color space used for this. If we have `OpaqueColor::<CS>::from_kelvin()`, we need to make sure the colors aren't chromatically adapted. As in, say we calculate the color temperature in sRGB as done here (sRGB having a D65 white point), and convert that to, say, ProPhoto RGB (with a D50 white point), the color is chromatically adapted, and now represents a different absolute color, with a different color temperature.

Reply:
This was discussed during office hours. The path forward is probably to put this behind a feature flag that also introduces a new wrapper type, `AbsoluteColor`, between which conversions can be done that don't do (or undo) the implicit chromatic adaptation performed by `ColorSpace::{to_linear_srgb,from_linear_srgb,convert}`. This requires storing the white point chromaticities on `ColorSpace` as an associated const (probably without a feature flag to maintain compatibility of the type between builds with different flags), and expose the white point on `DynamicColor` as well.